internal/common/pid_sock_status.go

```diff
@@ -42,19 +52,51 @@ func WriteAndHandleStatusFile() error {
 	if err != nil {
 		return fmt.Errorf("Failed to generate a random number for the unix socket path: '%w'.", err)
 	}
-	statusJson.UnixSocketPath = fmt.Sprintf("/tmp/autograder-%s.sock", unixFileNumber)
+	statusJson.UnixSocketPath = filepath.Join("/tmp", fmt.Sprintf("autograder-%s.sock", unixFileNumber))
 
 	err = util.ToJSONFile(statusJson, statusPath)
 	if err != nil {
-		return fmt.Errorf("Failed to write the pid to the status file: '%w'.", err)
+		return fmt.Errorf("Failed to write to the status file: '%w'.", err)
 	}
 
-	return nil
+	return err
 }
 
-func GetUnixSocketPath() (string, error) {
-	Lock(PID_SOCK_LOCK)
-	defer Unlock(PID_SOCK_LOCK)
+func checkAndHandleStalePid() (bool, error) {
+	statusPath := GetStatusPath()
+
+	if !util.IsFile(statusPath) {
+		return true, nil
+	}
+
+	var statusJson StatusInfo
+
```

Comment:
Remove this newline.
(Just a nit, but I think it is cleaner to declare it and fill it together.)